ext/SciMLBaseZygoteExt.jl

```diff
@@ -182,15 +191,15 @@ end
     NonlinearSolution{T, N, uType, R, P, A, O, uType2}(u, args...), NonlinearSolutionAdjoint
 end
 
-@adjoint function literal_getproperty(sol::AbstractTimeseriesSolution,
-        ::Val{:u})
-    function solu_adjoint(Δ)
-        zerou = zero(sol.prob.u0)
-        _Δ = @. ifelse(Δ === nothing, (zerou,), Δ)
-        (build_solution(sol.prob, sol.alg, sol.t, _Δ),)
-    end
-    sol.u, solu_adjoint
-end
+# @adjoint function literal_getproperty(sol::AbstractTimeseriesSolution,
+#         ::Val{:u})
+#     function solu_adjoint(Δ)
+#         zerou = zero(sol.prob.u0)
+#         _Δ = @. ifelse(Δ === nothing, (zerou,), Δ)
+#         (build_solution(sol.prob, sol.alg, sol.t, _Δ),)
+#     end
+#     sol.u, solu_adjoint
+# end
```

Comment:
Why is this removed?

Reply:
It was returning the `ODESolution` as the adjoint. It is also an issue because it shortcuts the gradients through parameters and instead replaces it with the `sol.prob`, whereas we need to accumulate the gradients here.

Reply:
Can you add a unit test in the downstream set which shows this is fine?

Reply:
Happy to. In fact, that's why I asked if anything was relying on this behavior previously. Could you suggest what kind of test you have in mind?

Reply:
I don't think this is what the error is referring to. I am missing a branch https://github.com/DhairyaLGandhi/RecursiveArrayTools.jl/tree/dg/noproj which removes an extra projection rule.

Reply:
It does refer to projecting to a `VectorOfArray`, and that rule wasn't defined for `Tangent`. Removing it gets us the expected results. If we want to project back to a `VectorOfArray` type, then that needs to be handled elsewhere.

Reply:
now that we have restored the adjoint, I believe this can be resolved